tutorials/platformer/activity4.md

```diff
@@ -0,0 +1,680 @@
+})
+function startNextLevel () {
+    for (let value of sprites.allOfKind(SpriteKind.Enemy)) {
+        value.destroy()
+    }
+    if (currentLevel == 0) {
+        tiles.setTilemap(tilemap`level`)
+    } else if (currentLevel == 1) {
+        tiles.setTilemap(tilemap`level`)
+    } else {
+        game.over(true)
+    }
+    tiles.placeOnRandomTile(thePlayer, myTiles.tile3)
+    for (let value of tiles.getTilesByType(myTiles.tile5)) {
+        newEnemy = sprites.create(img`
+            a a a a a a a a a a a a a a a a
+            a b b b b b b b b b b b b b b a
+            a b a a a a a a a a a a a a b a
+            a b a a b b a a a a b b a a b a
+            a b a a a a b a a b a a a a b a
+            a b a a a a a a a a a a a a b a
+            a b a a a b a a a a b a a a b a
+            a b a a a b a a a a b a a a b a
+            a b a a a a a a a a a a a a b a
+            a b a a a a a a a a a a a a b a
+            a b a a a b b b b b b a a a b a
+            a b a a b a a a a a a b a a b a
+            a b a a a a a a a a a a a a b a
+            a b a a a a a a a a a a a a b a
+            a b b b b b b b b b b b b b b a
+            a a a a a a a a a a a a a a a a
+            `, SpriteKind.Enemy)
+        tiles.placeOnTile(newEnemy, value)
+        newEnemy.follow(thePlayer, 30)
+    }
+    currentLevel += 1
+}
+sprites.onOverlap(SpriteKind.Player, SpriteKind.Enemy, function (sprite, otherSprite) {
+    otherSprite.destroy()
+    if (sprite.bottom < otherSprite.y) {
+        sprite.vy = -100
+    } else {
+        info.changeLifeBy(-1)
+    }
+})
+let newEnemy: Sprite = null
+let currentLevel = 0
+let thePlayer: Sprite = null
+scene.setBackgroundColor(11)
+thePlayer = sprites.create(img`
+    3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3
+    3 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3
+    3 1 3 3 3 3 3 3 3 3 3 3 3 3 1 3
+    3 1 3 3 3 3 3 3 3 3 3 3 3 3 1 3
+    3 1 3 3 3 3 3 3 3 3 3 3 3 3 1 3
+    3 1 3 3 1 1 1 3 3 3 1 3 3 3 1 3
+    3 1 3 3 1 3 3 1 3 1 1 3 3 3 1 3
+    3 1 3 3 1 3 3 1 3 3 1 3 3 3 1 3
+    3 1 3 3 1 1 1 3 3 3 1 3 3 3 1 3
+    3 1 3 3 1 3 3 3 3 3 1 3 3 3 1 3
+    3 1 3 3 1 3 3 3 3 1 1 1 3 3 1 3
+    3 1 3 3 3 3 3 3 3 3 3 3 3 3 1 3
+    3 1 3 3 3 3 3 3 3 3 3 3 3 3 1 3
+    3 1 3 3 3 3 3 3 3 3 3 3 3 3 1 3
+    3 1 1 1 1 1 1 1 1 1 1 1 1 1 1 3
+    3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3
+    `, SpriteKind.Player)
+thePlayer.ay = 500
+controller.moveSprite(thePlayer, 100, 0)
+scene.cameraFollowSprite(thePlayer)
+info.setLife(3)
+startNextLevel()
+
+
+```
+
+## Start
+
+In this lesson, we'll learn how to add more levels to our game.
+We'll also learn how to create a function so that we can reuse code in multiple places.
+
+## Make a function
+
+First, let's create a new function!
+In the toolbox, expand the "Advanced" tab and click on the "Functions" category.
+Press the "Make a Function..." button to create a new function.
+Name this function "startNextLevel".
+
+```blocks
+function startNextLevel() {
+}
+```
+
+
+## Refactoring pt. 1
```

Comment:
i know this is not displayed to the user but maybe "reorganizing" in case someone views the markdown/docs?